{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Building a Drug Response Report\n",
    "\n",
    "For this notebook, you are going to focus on a single region in the genome, defined as chromosome 22, for all 2,504 samples in the Thousand Genomes dataset. As chromosome 22 was the first chromosome to be sequenced as part of the Human Genome Project, it is your first here as well. \n",
    "\n",
    "What small molecules/drugs are most likely to affect a subpopulation of individuals (ancestry, age, etc.) based on their genomic information?\n",
    "\n",
    "In this query, assume that you have some phenotype data about your population. In this case, also assume that all samples sharing the pattern “NA12” are part of a specific demographic.\n",
    "\n",
    "In this query, use sampleid as your predicate pushdown. The general steps are:\n",
    "\n",
    "1. Filter by the samples in your subpopulation\n",
    "2. Aggregate variant frequencies for the subpopulation-of-interest\n",
    "3. Join on ClinVar dataset\n",
    "4. Filter by variants that have been implicated in drug-response\n",
    "5. Order by highest frequency variants\n",
    "\n",
    "The raw clinvar data and a parquet version of chromosome 22 of 1000 genomes, partitioned by sample id, are in your data lake. You also have a VCF in your data lake for chromosome 22 of 1000 genomes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "import boto3, os\n",
    "\n",
    "s3 = boto3.resource('s3')\n",
    "glue = boto3.client('glue')\n",
    "cfn = boto3.client('cloudformation')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Looking in indexes: https://pypi.org/simple, https://pip.repos.neuron.amazonaws.com\n",
      "Requirement already satisfied: PyAthena in /home/ec2-user/anaconda3/envs/python3/lib/python3.8/site-packages (2.9.3)\n",
      "Requirement already satisfied: boto3>=1.4.4 in /home/ec2-user/anaconda3/envs/python3/lib/python3.8/site-packages (from PyAthena) (1.23.10)\n",
      "Requirement already satisfied: botocore>=1.5.52 in /home/ec2-user/anaconda3/envs/python3/lib/python3.8/site-packages (from PyAthena) (1.26.10)\n",
      "Requirement already satisfied: tenacity>=4.1.0 in /home/ec2-user/anaconda3/envs/python3/lib/python3.8/site-packages (from PyAthena) (8.0.1)\n",
      "Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /home/ec2-user/anaconda3/envs/python3/lib/python3.8/site-packages (from boto3>=1.4.4->PyAthena) (0.10.0)\n",
      "Requirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /home/ec2-user/anaconda3/envs/python3/lib/python3.8/site-packages (from boto3>=1.4.4->PyAthena) (0.5.2)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.25.4 in /home/ec2-user/anaconda3/envs/python3/lib/python3.8/site-packages (from botocore>=1.5.52->PyAthena) (1.26.8)\n",
      "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /home/ec2-user/anaconda3/envs/python3/lib/python3.8/site-packages (from botocore>=1.5.52->PyAthena) (2.8.2)\n",
      "Requirement already satisfied: six>=1.5 in /home/ec2-user/anaconda3/envs/python3/lib/python3.8/site-packages (from python-dateutil<3.0.0,>=2.1->botocore>=1.5.52->PyAthena) (1.16.0)\n",
      "\u001b[33mWARNING: You are using pip version 22.0.4; however, version 22.1.2 is available.\n",
      "You should consider upgrading via the '/home/ec2-user/anaconda3/envs/python3/bin/python -m pip install --upgrade pip' command.\u001b[0m\u001b[33m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "!{sys.executable} -m pip install PyAthena"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pyathena import connect\n",
    "import pandas as pd\n",
    "from pyathena.pandas.util import as_pandas"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define Variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "us-west-1\n",
      "GenomicsAnalysis\n",
      "genomicsanalysis\n",
      "genomicsanalysis-us-west-1\n",
      "genomicsanalysis-pipeline-datalakebucket-1g5hynzwz9kmd\n"
     ]
    }
   ],
   "source": [
    "import jmespath\n",
    "\n",
    "session = boto3.session.Session()\n",
    "region = session.region_name\n",
    "print(region)\n",
    "\n",
    "project_name = os.environ.get('RESOURCE_PREFIX')\n",
    "database_name = project_name.lower()\n",
    "work_group_name = project_name.lower() + '-' + region\n",
    "print(project_name)\n",
    "print(database_name)\n",
    "print(work_group_name)\n",
    "\n",
    "resources = cfn.describe_stacks(StackName='{0}-Pipeline'.format(project_name))\n",
    "query = 'Stacks[].Outputs[?OutputKey==`DataLakeBucket`].OutputValue'\n",
    "data_lake_bucket = path = jmespath.search(query, resources)[0][0]\n",
    "print(data_lake_bucket)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create drug response result set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genotypefrequency</th>\n",
       "      <th>rsid</th>\n",
       "      <th>phenotypelist</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>startposition</th>\n",
       "      <th>endposition</th>\n",
       "      <th>referenceallele</th>\n",
       "      <th>alternateallele</th>\n",
       "      <th>genotype0</th>\n",
       "      <th>genotype1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.553846</td>\n",
       "      <td>4680</td>\n",
       "      <td>CATECHOL-O-METHYLTRANSFERASE POLYMORPHISM;meth...</td>\n",
       "      <td>22</td>\n",
       "      <td>19951270</td>\n",
       "      <td>19951271</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.507692</td>\n",
       "      <td>2298383</td>\n",
       "      <td>caffeine response - Toxicity/ADR</td>\n",
       "      <td>22</td>\n",
       "      <td>24825510</td>\n",
       "      <td>24825511</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.369231</td>\n",
       "      <td>3892097</td>\n",
       "      <td>Debrisoquine, poor metabolism of;amitriptyline...</td>\n",
       "      <td>22</td>\n",
       "      <td>42524946</td>\n",
       "      <td>42524947</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.353846</td>\n",
       "      <td>2298383</td>\n",
       "      <td>caffeine response - Toxicity/ADR</td>\n",
       "      <td>22</td>\n",
       "      <td>24825510</td>\n",
       "      <td>24825511</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.292308</td>\n",
       "      <td>738409</td>\n",
       "      <td>Fatty liver disease, nonalcoholic 1;Susceptibi...</td>\n",
       "      <td>22</td>\n",
       "      <td>44324726</td>\n",
       "      <td>44324727</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.276923</td>\n",
       "      <td>13306278</td>\n",
       "      <td>Selective serotonin reuptake inhibitors respon...</td>\n",
       "      <td>22</td>\n",
       "      <td>19929026</td>\n",
       "      <td>19929027</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.153846</td>\n",
       "      <td>4680</td>\n",
       "      <td>CATECHOL-O-METHYLTRANSFERASE POLYMORPHISM;meth...</td>\n",
       "      <td>22</td>\n",
       "      <td>19951270</td>\n",
       "      <td>19951271</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.046154</td>\n",
       "      <td>738409</td>\n",
       "      <td>Fatty liver disease, nonalcoholic 1;Susceptibi...</td>\n",
       "      <td>22</td>\n",
       "      <td>44324726</td>\n",
       "      <td>44324727</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.046154</td>\n",
       "      <td>13306278</td>\n",
       "      <td>Selective serotonin reuptake inhibitors respon...</td>\n",
       "      <td>22</td>\n",
       "      <td>19929026</td>\n",
       "      <td>19929027</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.030769</td>\n",
       "      <td>3892097</td>\n",
       "      <td>Debrisoquine, poor metabolism of;amitriptyline...</td>\n",
       "      <td>22</td>\n",
       "      <td>42524946</td>\n",
       "      <td>42524947</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   genotypefrequency      rsid  \\\n",
       "0           0.553846      4680   \n",
       "1           0.507692   2298383   \n",
       "2           0.369231   3892097   \n",
       "3           0.353846   2298383   \n",
       "4           0.292308    738409   \n",
       "5           0.276923  13306278   \n",
       "6           0.153846      4680   \n",
       "7           0.046154    738409   \n",
       "8           0.046154  13306278   \n",
       "9           0.030769   3892097   \n",
       "\n",
       "                                       phenotypelist chromosome  \\\n",
       "0  CATECHOL-O-METHYLTRANSFERASE POLYMORPHISM;meth...         22   \n",
       "1                   caffeine response - Toxicity/ADR         22   \n",
       "2  Debrisoquine, poor metabolism of;amitriptyline...         22   \n",
       "3                   caffeine response - Toxicity/ADR         22   \n",
       "4  Fatty liver disease, nonalcoholic 1;Susceptibi...         22   \n",
       "5  Selective serotonin reuptake inhibitors respon...         22   \n",
       "6  CATECHOL-O-METHYLTRANSFERASE POLYMORPHISM;meth...         22   \n",
       "7  Fatty liver disease, nonalcoholic 1;Susceptibi...         22   \n",
       "8  Selective serotonin reuptake inhibitors respon...         22   \n",
       "9  Debrisoquine, poor metabolism of;amitriptyline...         22   \n",
       "\n",
       "   startposition  endposition referenceallele alternateallele genotype0  \\\n",
       "0       19951270     19951271               G               A         0   \n",
       "1       24825510     24825511               C               T         0   \n",
       "2       42524946     42524947               C               T         0   \n",
       "3       24825510     24825511               C               T         1   \n",
       "4       44324726     44324727               C               G         0   \n",
       "5       19929026     19929027               C               T         0   \n",
       "6       19951270     19951271               G               A         1   \n",
       "7       44324726     44324727               C               G         1   \n",
       "8       19929026     19929027               C               T         1   \n",
       "9       42524946     42524947               C               T         1   \n",
       "\n",
       "  genotype1  \n",
       "0         1  \n",
       "1         1  \n",
       "2         1  \n",
       "3         1  \n",
       "4         1  \n",
       "5         1  \n",
       "6         1  \n",
       "7         1  \n",
       "8         1  \n",
       "9         1  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conn = connect(s3_staging_dir='s3://%s/results/drug_response' % data_lake_bucket, region_name=region)\n",
    "cursor = conn.cursor(work_group=work_group_name)\n",
    "cursor.execute(\n",
    "        \"SELECT\"\n",
    "        \"    count(*)/cast(numsamples AS DOUBLE) AS genotypefrequency \"\n",
    "        \"    ,cv.rsid \"\n",
    "        \"    ,cv.phenotypelist \"\n",
    "        \"    ,sv.chromosome \"\n",
    "        \"    ,sv.startposition \"\n",
    "        \"    ,sv.endposition \"\n",
    "        \"    ,sv.referenceallele \"\n",
    "        \"    ,sv.alternateallele \"\n",
    "        \"    ,sv.genotype0 \"\n",
    "        \"    ,sv.genotype1 \"\n",
    "        \"FROM \\\"\" + database_name + \"\\\".onekg_chr22_by_sample sv \"\n",
    "        \"CROSS JOIN \"\n",
    "        \"    (SELECT count(1) AS numsamples \"\n",
    "        \"    FROM \"\n",
    "        \"        (SELECT DISTINCT sampleid \"\n",
    "        \"        FROM \\\"\" + database_name + \"\\\".onekg_chr22_by_sample \"\n",
    "        \"        WHERE sampleid LIKE 'NA12%')) \"\n",
    "        \"JOIN \\\"\" + database_name + \"\\\".clinvar cv \"\n",
    "        \"ON sv.chromosome = cv.chromosome \"\n",
    "        \"    AND sv.startposition = cv.start - 1 \"\n",
    "        \"    AND sv.endposition = cv.stop \"\n",
    "        \"    AND sv.referenceallele = cv.referenceallele \"\n",
    "        \"    AND sv.alternateallele = cv.alternateallele \"\n",
    "        \"WHERE assembly='GRCh37' \"\n",
    "        \"    AND cv.clinicalsignificance LIKE '%response%' \"\n",
    "        \"    AND sampleid LIKE 'NA12%' \"\n",
    "        \"GROUP BY  sv.chromosome \"\n",
    "        \"          ,sv.startposition \"\n",
    "        \"          ,sv.endposition \"\n",
    "        \"          ,sv.referenceallele \"\n",
    "        \"          ,sv.alternateallele \"\n",
    "        \"          ,sv.genotype0 \"\n",
    "        \"          ,sv.genotype1 \"\n",
    "        \"          ,cv.clinicalsignificance \"\n",
    "        \"          ,cv.phenotypelist \"\n",
    "        \"          ,cv.rsid \"\n",
    "        \"          ,numsamples \"\n",
    "        \"ORDER BY  genotypefrequency DESC LIMIT 50 \")\n",
    "\n",
    "df = as_pandas(cursor)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Query annotation dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alleleid</th>\n",
       "      <th>type</th>\n",
       "      <th>name</th>\n",
       "      <th>geneid</th>\n",
       "      <th>genesymbol</th>\n",
       "      <th>hgnc_id</th>\n",
       "      <th>clinicalsignificance</th>\n",
       "      <th>clinsigsimple</th>\n",
       "      <th>lastevaluated</th>\n",
       "      <th>rsid</th>\n",
       "      <th>...</th>\n",
       "      <th>referenceallele</th>\n",
       "      <th>alternateallele</th>\n",
       "      <th>cytogenetic</th>\n",
       "      <th>reviewstatus</th>\n",
       "      <th>numbersubmitters</th>\n",
       "      <th>guidelines</th>\n",
       "      <th>testedingtr</th>\n",
       "      <th>otherids</th>\n",
       "      <th>submittercategories</th>\n",
       "      <th>variationid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15041</td>\n",
       "      <td>indel</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704705</td>\n",
       "      <td>...</td>\n",
       "      <td>GGAT</td>\n",
       "      <td>TGCTGTAAACTGTAACTGTAAA</td>\n",
       "      <td>7p22.1</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM Allelic Variant:613653.0001</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>15041</td>\n",
       "      <td>indel</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704705</td>\n",
       "      <td>...</td>\n",
       "      <td>GGAT</td>\n",
       "      <td>TGCTGTAAACTGTAACTGTAAA</td>\n",
       "      <td>7p22.1</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM Allelic Variant:613653.0001</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15042</td>\n",
       "      <td>deletion</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704709</td>\n",
       "      <td>...</td>\n",
       "      <td>GGACCTGCCCTGCT</td>\n",
       "      <td>-</td>\n",
       "      <td>7p22.1</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM Allelic Variant:613653.0002</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15042</td>\n",
       "      <td>deletion</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704709</td>\n",
       "      <td>...</td>\n",
       "      <td>GGACCTGCCCTGCT</td>\n",
       "      <td>-</td>\n",
       "      <td>7p22.1</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM Allelic Variant:613653.0002</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15043</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
       "      <td>9640</td>\n",
       "      <td>ZNF592</td>\n",
       "      <td>HGNC:28986</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Jun 29, 2015</td>\n",
       "      <td>150829393</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>15q25</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM Allelic Variant:613624.0001,UniProtKB (pr...</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>15043</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
       "      <td>9640</td>\n",
       "      <td>ZNF592</td>\n",
       "      <td>HGNC:28986</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Jun 29, 2015</td>\n",
       "      <td>150829393</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>15q25.3</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM Allelic Variant:613624.0001,UniProtKB (pr...</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>15044</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 07, 2017</td>\n",
       "      <td>267606829</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>11q24</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM Allelic Variant:613622.0001</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>15044</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 07, 2017</td>\n",
       "      <td>267606829</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>11q24.2</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM Allelic Variant:613622.0001</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>15045</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Oct 01, 2010</td>\n",
       "      <td>267606830</td>\n",
       "      <td>...</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>11q24</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM Allelic Variant:613622.0002,UniProtKB (pr...</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>15045</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n",
       "      <td>55572</td>\n",
       "      <td>FOXRED1</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Oct 01, 2010</td>\n",
       "      <td>267606830</td>\n",
       "      <td>...</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>11q24.2</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM Allelic Variant:613622.0002,UniProtKB (pr...</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  alleleid                       type  \\\n",
       "0    15041                      indel   \n",
       "1    15041                      indel   \n",
       "2    15042                   deletion   \n",
       "3    15042                   deletion   \n",
       "4    15043  single nucleotide variant   \n",
       "5    15043  single nucleotide variant   \n",
       "6    15044  single nucleotide variant   \n",
       "7    15044  single nucleotide variant   \n",
       "8    15045  single nucleotide variant   \n",
       "9    15045  single nucleotide variant   \n",
       "\n",
       "                                                name geneid genesymbol  \\\n",
       "0  NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...   9907      AP5Z1   \n",
       "1  NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...   9907      AP5Z1   \n",
       "2     NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)   9907      AP5Z1   \n",
       "3     NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)   9907      AP5Z1   \n",
       "4       NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)   9640     ZNF592   \n",
       "5       NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)   9640     ZNF592   \n",
       "6        NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)  55572    FOXRED1   \n",
       "7        NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)  55572    FOXRED1   \n",
       "8       NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)  55572    FOXRED1   \n",
       "9       NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)  55572    FOXRED1   \n",
       "\n",
       "      hgnc_id    clinicalsignificance clinsigsimple lastevaluated       rsid  \\\n",
       "0  HGNC:22197              Pathogenic             1  Jun 29, 2010  397704705   \n",
       "1  HGNC:22197              Pathogenic             1  Jun 29, 2010  397704705   \n",
       "2  HGNC:22197              Pathogenic             1  Jun 29, 2010  397704709   \n",
       "3  HGNC:22197              Pathogenic             1  Jun 29, 2010  397704709   \n",
       "4  HGNC:28986  Uncertain significance             0  Jun 29, 2015  150829393   \n",
       "5  HGNC:28986  Uncertain significance             0  Jun 29, 2015  150829393   \n",
       "6  HGNC:26927              Pathogenic             1  Dec 07, 2017  267606829   \n",
       "7  HGNC:26927              Pathogenic             1  Dec 07, 2017  267606829   \n",
       "8  HGNC:26927              Pathogenic             1  Oct 01, 2010  267606830   \n",
       "9  HGNC:26927              Pathogenic             1  Oct 01, 2010  267606830   \n",
       "\n",
       "   ... referenceallele         alternateallele cytogenetic  \\\n",
       "0  ...            GGAT  TGCTGTAAACTGTAACTGTAAA      7p22.1   \n",
       "1  ...            GGAT  TGCTGTAAACTGTAACTGTAAA      7p22.1   \n",
       "2  ...  GGACCTGCCCTGCT                       -      7p22.1   \n",
       "3  ...  GGACCTGCCCTGCT                       -      7p22.1   \n",
       "4  ...               G                       A       15q25   \n",
       "5  ...               G                       A     15q25.3   \n",
       "6  ...               C                       T       11q24   \n",
       "7  ...               C                       T     11q24.2   \n",
       "8  ...               A                       G       11q24   \n",
       "9  ...               A                       G     11q24.2   \n",
       "\n",
       "                          reviewstatus numbersubmitters guidelines  \\\n",
       "0       no assertion criteria provided                1       None   \n",
       "1       no assertion criteria provided                1       None   \n",
       "2       no assertion criteria provided                1       None   \n",
       "3       no assertion criteria provided                1       None   \n",
       "4       no assertion criteria provided                1       None   \n",
       "5       no assertion criteria provided                1       None   \n",
       "6  criteria provided, single submitter                2       None   \n",
       "7  criteria provided, single submitter                2       None   \n",
       "8       no assertion criteria provided                1       None   \n",
       "9       no assertion criteria provided                1       None   \n",
       "\n",
       "  testedingtr                                           otherids  \\\n",
       "0           N                   OMIM Allelic Variant:613653.0001   \n",
       "1           N                   OMIM Allelic Variant:613653.0001   \n",
       "2           N                   OMIM Allelic Variant:613653.0002   \n",
       "3           N                   OMIM Allelic Variant:613653.0002   \n",
       "4           N  OMIM Allelic Variant:613624.0001,UniProtKB (pr...   \n",
       "5           N  OMIM Allelic Variant:613624.0001,UniProtKB (pr...   \n",
       "6           N                   OMIM Allelic Variant:613622.0001   \n",
       "7           N                   OMIM Allelic Variant:613622.0001   \n",
       "8           N  OMIM Allelic Variant:613622.0002,UniProtKB (pr...   \n",
       "9           N  OMIM Allelic Variant:613622.0002,UniProtKB (pr...   \n",
       "\n",
       "  submittercategories  variationid  \n",
       "0                   1            2  \n",
       "1                   1            2  \n",
       "2                   1            3  \n",
       "3                   1            3  \n",
       "4                   1            4  \n",
       "5                   1            4  \n",
       "6                   3            5  \n",
       "7                   3            5  \n",
       "8                   1            6  \n",
       "9                   1            6  \n",
       "\n",
       "[10 rows x 31 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conn = connect(s3_staging_dir='s3://%s/results/annotation/clinvar' % data_lake_bucket,region_name=region)\n",
    "cursor = conn.cursor(work_group=work_group_name)\n",
    "cursor.execute('SELECT * FROM \"%s\".\"clinvar\" limit 10' % database_name)\n",
    "df = as_pandas(cursor)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Query cohort dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alternateallele</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>endposition</th>\n",
       "      <th>genotype0</th>\n",
       "      <th>genotype1</th>\n",
       "      <th>referenceallele</th>\n",
       "      <th>sampleid</th>\n",
       "      <th>startposition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>32726515</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>NA12283</td>\n",
       "      <td>32726514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>32726515</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>NA12750</td>\n",
       "      <td>32726514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>32726515</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>NA12872</td>\n",
       "      <td>32726514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>T</td>\n",
       "      <td>22</td>\n",
       "      <td>32726587</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>NA12046</td>\n",
       "      <td>32726586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>T</td>\n",
       "      <td>22</td>\n",
       "      <td>32726587</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>NA12383</td>\n",
       "      <td>32726586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>T</td>\n",
       "      <td>22</td>\n",
       "      <td>32726587</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>NA12777</td>\n",
       "      <td>32726586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>32726606</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>NA12283</td>\n",
       "      <td>32726605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>32726606</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>NA12750</td>\n",
       "      <td>32726605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>32726606</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>NA12872</td>\n",
       "      <td>32726605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>32726649</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>GGT</td>\n",
       "      <td>NA12814</td>\n",
       "      <td>32726646</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  alternateallele chromosome  endposition genotype0 genotype1 referenceallele  \\\n",
       "0               G         22     32726515         0         1               C   \n",
       "1               G         22     32726515         1         1               C   \n",
       "2               G         22     32726515         1         1               C   \n",
       "3               T         22     32726587         1         1               A   \n",
       "4               T         22     32726587         1         1               A   \n",
       "5               T         22     32726587         1         1               A   \n",
       "6               G         22     32726606         1         1               A   \n",
       "7               G         22     32726606         1         1               A   \n",
       "8               G         22     32726606         1         1               A   \n",
       "9               G         22     32726649         0         1             GGT   \n",
       "\n",
       "  sampleid  startposition  \n",
       "0  NA12283       32726514  \n",
       "1  NA12750       32726514  \n",
       "2  NA12872       32726514  \n",
       "3  NA12046       32726586  \n",
       "4  NA12383       32726586  \n",
       "5  NA12777       32726586  \n",
       "6  NA12283       32726605  \n",
       "7  NA12750       32726605  \n",
       "8  NA12872       32726605  \n",
       "9  NA12814       32726646  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conn = connect(s3_staging_dir='s3://%s/results/variants/' % data_lake_bucket,region_name=region)\n",
    "cursor = conn.cursor(work_group=work_group_name)\n",
    "cursor.execute(\"SELECT * FROM \\\"\" + database_name + \"\\\".onekg_chr22_by_sample WHERE sampleid LIKE 'NA12%' limit 10\")\n",
    "df = as_pandas(cursor)\n",
    "df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Query individual variant dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>v.contig</th>\n",
       "      <th>v.start</th>\n",
       "      <th>v.ref</th>\n",
       "      <th>v.altalleles</th>\n",
       "      <th>va.rsid</th>\n",
       "      <th>va.qual</th>\n",
       "      <th>va.filters</th>\n",
       "      <th>va.info.end</th>\n",
       "      <th>va.info.blockavg_min30p3a</th>\n",
       "      <th>va.info.snvhpol</th>\n",
       "      <th>va.info.cigar</th>\n",
       "      <th>va.info.ru</th>\n",
       "      <th>va.info.refrep</th>\n",
       "      <th>va.info.idrep</th>\n",
       "      <th>va.info.mq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14776</td>\n",
       "      <td>G</td>\n",
       "      <td>[{ref=G, alt=A}]</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[LowGQX]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14794</td>\n",
       "      <td>G</td>\n",
       "      <td>[{ref=G, alt=C}]</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[LowDepth, LowGQX]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16014</td>\n",
       "      <td>C</td>\n",
       "      <td>[{ref=C, alt=T}]</td>\n",
       "      <td>None</td>\n",
       "      <td>5.0</td>\n",
       "      <td>[LowDepth, LowGQX]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16674</td>\n",
       "      <td>G</td>\n",
       "      <td>[{ref=G, alt=T}]</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[LowGQX]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16866</td>\n",
       "      <td>G</td>\n",
       "      <td>[{ref=G, alt=T}]</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>[LowDepth, LowGQX]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16957</td>\n",
       "      <td>G</td>\n",
       "      <td>[{ref=G, alt=T}]</td>\n",
       "      <td>None</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[LowDepth, LowGQX]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17004</td>\n",
       "      <td>C</td>\n",
       "      <td>[{ref=C, alt=G}]</td>\n",
       "      <td>None</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[LowDepth, LowGQX]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17006</td>\n",
       "      <td>A</td>\n",
       "      <td>[{ref=A, alt=G}]</td>\n",
       "      <td>None</td>\n",
       "      <td>3.0</td>\n",
       "      <td>[LowDepth, LowGQX]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17259</td>\n",
       "      <td>T</td>\n",
       "      <td>[{ref=T, alt=C}]</td>\n",
       "      <td>None</td>\n",
       "      <td>0.0</td>\n",
       "      <td>[LowGQX]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17365</td>\n",
       "      <td>C</td>\n",
       "      <td>[{ref=C, alt=G}]</td>\n",
       "      <td>None</td>\n",
       "      <td>17.0</td>\n",
       "      <td>[LowGQX]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  v.contig  v.start v.ref      v.altalleles va.rsid  va.qual  \\\n",
       "0     chr1    14776     G  [{ref=G, alt=A}]    None      0.0   \n",
       "1     chr1    14794     G  [{ref=G, alt=C}]    None      0.0   \n",
       "2     chr1    16014     C  [{ref=C, alt=T}]    None      5.0   \n",
       "3     chr1    16674     G  [{ref=G, alt=T}]    None      1.0   \n",
       "4     chr1    16866     G  [{ref=G, alt=T}]    None      1.0   \n",
       "5     chr1    16957     G  [{ref=G, alt=T}]    None      3.0   \n",
       "6     chr1    17004     C  [{ref=C, alt=G}]    None      3.0   \n",
       "7     chr1    17006     A  [{ref=A, alt=G}]    None      3.0   \n",
       "8     chr1    17259     T  [{ref=T, alt=C}]    None      0.0   \n",
       "9     chr1    17365     C  [{ref=C, alt=G}]    None     17.0   \n",
       "\n",
       "           va.filters va.info.end va.info.blockavg_min30p3a  va.info.snvhpol  \\\n",
       "0            [LowGQX]        None                      None                3   \n",
       "1  [LowDepth, LowGQX]        None                      None                5   \n",
       "2  [LowDepth, LowGQX]        None                      None                2   \n",
       "3            [LowGQX]        None                      None                3   \n",
       "4  [LowDepth, LowGQX]        None                      None                2   \n",
       "5  [LowDepth, LowGQX]        None                      None                2   \n",
       "6  [LowDepth, LowGQX]        None                      None                3   \n",
       "7  [LowDepth, LowGQX]        None                      None                2   \n",
       "8            [LowGQX]        None                      None                2   \n",
       "9            [LowGQX]        None                      None                4   \n",
       "\n",
       "  va.info.cigar va.info.ru va.info.refrep va.info.idrep  va.info.mq  \n",
       "0          None       None           None          None         7.0  \n",
       "1          None       None           None          None         6.0  \n",
       "2          None       None           None          None         7.0  \n",
       "3          None       None           None          None         7.0  \n",
       "4          None       None           None          None         4.0  \n",
       "5          None       None           None          None         3.0  \n",
       "6          None       None           None          None         4.0  \n",
       "7          None       None           None          None         4.0  \n",
       "8          None       None           None          None        15.0  \n",
       "9          None       None           None          None        14.0  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conn = connect(s3_staging_dir='s3://%s/results/vcf/' % data_lake_bucket,region_name=region)\n",
    "cursor = conn.cursor(work_group=work_group_name)\n",
    "cursor.execute(\"SELECT * FROM \\\"\" + database_name + \"\\\".vcf limit 10\")\n",
    "df = as_pandas(cursor)\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Raw Cell Format",
  "kernelspec": {
   "display_name": "conda_python3",
   "language": "python",
   "name": "conda_python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
